### PR TITLE
Add gh-pages deployment for the docs

### PR DESCRIPTION
**Pull request by jaltekruse**
I have added a new github action that deploys the docs on commits to the main branch.

It won't run for this PR until it is merged by design, but you can view a test of this getting deployed for a version I pushed to the main branch on my fork.

https://jaltekruse.github.io/DoenetML

### CODE REVIEW

File: packages/docs-nextra/next.config.mjs [hunk 1]
```diff
@@ -87,7 +87,13 @@ const withNextra = nextraConfig({
 //     latex: true
 //   });
 
-export default withNextra();
+export default withNextra({
+    output: "export",
+    basePath: '/DoenetML',
```
[the remark this answers is not in the document]
Comment (jaltekruse) [reply packages/docs-nextra/next.config.mjs]: Thanks, I should have also done a local test after making the changes, applied this change to the branch.

**Comment by siefkenj** [comment]
Looks like a modification to `iframe-doenet` needs to be done. See this page: https://jaltekruse.github.io/DoenetML/tutorials/intermediateTutorial/iT2-2-points  it is looking for `“https://jaltekruse.github.io/bundle/doenet-standalone.js”` which doesn't exist. Somehow we need to percolate the prefix down to the iframe component.

**Comment by siefkenj** [comment]
Now that the new doenet iframe component is been created, we can use it instead. It should be more robust :-)

**Comment by siefkenj** [comment]
@jaltekruse Looks like you have some lints to fix

**Pull request merged**